### ISSUE TITLE
Documentation for how to set user attributes

### ISSUE THREAD

**Issue by ptrwtts** [intuit/wasabi#204]
Hi, we're evaluating using Wasabi, and there's one thing I haven't been able to figure out from the docs. If we want to limit an experiment to users from CA for example, how do we let Wasabi know which users these are? Is it:

a) Set it in advance, via an API call?
b) Looked up from our database during assignment?
c) Passed from the client in the assignment API call?

While digging through the Java docs I see mention of a "SegmentationProfile" and related mentions of "Hyrule", but haven't been able to figure out how these are set. Is there API documentation somewhere?

**Comment by longdogz** [comment]
Sorry that this wasn't documented better.  The first thing you might want to know is that, if you are able to get Wasabi installed on your local machine, you can get to the API docs (using Swagger) by cd'ing to your local wasabi repo and typing:

`./bin/wasabi.sh resource:api`

This is mentioned toward the end of the README (scroll down when you're on the top level wasabi page here on github).

But to answer your question, Wasabi definitely handles segmenting users, e.g., limiting the users put in the test by the state they live in.  You can control which users are in the test based on their passing a segmentation rule.  You define the rule using the Wasabi Admin UI, in the Segmentation tab when you are editing an experiment.  You can either define it using widgets (if your rule is simple enough, e.g., the example you gave) or you can define it using a rule language (the online help has some information about the language).  Usually, you will be fine using the widgets and then you don't need to remember the language syntax.

In order to actually use segmentation in Wasabi, you create an experiment and define a segmentation rule for it.  Part of what you define in the segmentation rule is the name of the parameters that will contain the value for each user, e.g., "state" in your example.  Then you start your experiment.  When you call the Wasabi to request whether a particular user is in the experiment and, if so, what bucket they're assigned to, you need to call a version of the API where you pass in some JSON that includes "profile information" about the user, that is, things like what state they are in.  Look at the API Calls tab for your experiment in the Admin UI for an example call using curl.  The information you pass is then plugged into the rule and if the result is "true", then the user is allowed in the test.  Of course, then things like sampling rate, mutual exclusion, etc. will come into play, but at least the user will now be past the segmentation stage.  If they do not pass the segmentation rule, you will get that in the response from the API call.

So, basically, the answer is c) in your list of choices in your question.

Note that Hyrule is simply the engine we use to evaluate rules.  It is what executes the language using the profile.

**Comment by ptrwtts** [comment]
Thanks for the prompt reply! That makes sense, and is what I suspected. Looking forward to doing a full investigation and kicking the tires on it very soon. Cheers!

**Comment by alexanderdean** [comment]
Was this one closed without updating the documentation? It's still a bit confusing. Suggestions:

1. Add a section to the user-guide on this with a screenshot of the API Calls section
2. Provide a more complex `curl` example to the `/assignments` endpoint in the "Call Wasabi" section of the user guide

**Comment by kevinwongolx** [comment]
I was trying to solve exactly the same problem, then I stumbled upon this post
https://github.com/intuit/wasabi/issues/289
